scripts/upgrade

```diff
@@ -30,13 +30,13 @@ if [[ -z "${secret_key:-}" ]]; then
 fi
 
 #=================================================
-# BACKUP SQLITE DATABASE
+# BACKUP SQLITE DATABASE AND DATAS
 #=================================================
 ynh_script_progression --message="Backup of the DB in temporary file...." --weight=5
 
-if [ -f "$install_dir/data/association.sqlite" ];then
-cp -a $install_dir/data/association.sqlite /tmp/
-fi
+# copy data directory backup
+file=$(mktemp -d $app_dataXXX --suffix=BKP)
+cp -a $install_dir/data /var/www/$file
```

Comment:
uuugh okay but `mktemp -d` already *is* a directory, you should copy stuff directly to that location ... not `/var/www/$(mktemp -d)`

also considering that `mktemp -d` returns a directory, it would be convenient to not call it `$file` but something like `$tmp_data_backup`

Finally, it would be good to consistently use `rsync` : either use it for both operations, or use only `cp`, but not both